src/redisearch_rs/inverted_index/src/numeric/mod.rs

```diff
@@ -0,0 +1,624 @@
+                    IoSlice::new(bytes),
+                ])?
+            }
+            Value::Float32Positive(value) => {
+                let bytes = value.to_le_bytes();
+
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::Float32Positive,
+                };
+
+                writer.write_vectored_all(&mut [
+                    IoSlice::new(&header.pack()),
+                    IoSlice::new(delta),
+                    IoSlice::new(&bytes),
+                ])?
+            }
+            Value::Float32Negative(value) => {
+                let bytes = value.to_le_bytes();
+
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::Float32Negative,
+                };
+
+                writer.write_vectored_all(&mut [
+                    IoSlice::new(&header.pack()),
+                    IoSlice::new(delta),
+                    IoSlice::new(&bytes),
+                ])?
+            }
+            Value::Float64Positive(value) => {
+                let bytes = value.to_le_bytes();
+
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::Float64Positive,
+                };
+
+                writer.write_vectored_all(&mut [
+                    IoSlice::new(&header.pack()),
+                    IoSlice::new(delta),
+                    IoSlice::new(&bytes),
+                ])?
+            }
+            Value::Float64Negative(value) => {
+                let bytes = value.to_le_bytes();
+
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::Float64Negative,
+                };
+
+                writer.write_vectored_all(&mut [
+                    IoSlice::new(&header.pack()),
+                    IoSlice::new(delta),
+                    IoSlice::new(&bytes),
+                ])?
+            }
+            Value::FloatInfinity => {
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::FloatInfinite,
+                };
+
+                writer
+                    .write_vectored_all(&mut [IoSlice::new(&header.pack()), IoSlice::new(delta)])?
+            }
+            Value::FloatNegInfinity => {
+                let header = Header {
+                    delta_bytes,
+                    typ: HeaderType::FloatNegInfinite,
+                };
+
+                writer
+                    .write_vectored_all(&mut [IoSlice::new(&header.pack()), IoSlice::new(delta)])?
+            }
+        };
+
+        let bytes_written = writer.stream_position()? - writer_len;
+
+        Ok(bytes_written as _)
+    }
+}
+
+impl Decoder for Numeric {
+    fn decode<R: Read>(
+        &self,
+        reader: &mut R,
+        base: t_docId,
+    ) -> std::io::Result<Option<DecoderResult>> {
+        let mut header = [0; 1];
+        reader.read_exact(&mut header)?;
```

Comment:
Shouldn't we check here if the reader is empty and return `Ok(None)` accordingly?

See:

```rust
    /// Decode the next record from the reader. If any delta values are decoded, then they should
    /// add to the `base` document ID to get the actual document ID.
    ///
    /// Returns `Ok(None)` if there is nothing left on the reader to decode.
    fn decode<R: Read>(
        &self,
        reader: &mut R,
        base: t_docId,
    ) -> std::io::Result<Option<DecoderResult>>;
```